packages/client/src/components/ProductListDropDown/DropDownViewList/DropDownViewList.component.js

```diff
@@ -1,39 +1,59 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { apiURL } from '../../../apiURL';
 import DropDownView from '../DropDownView/DropDownView.component';
 import './DropDownViewList.style.css';
 
-const DropDownViewList = () => {
-  const Categories = [
-    'Categories',
-    'Category 1',
-    'Category 2',
-    'Category 3',
-    'Category 4',
-    'Category 5',
-  ];
-  const SortBy = [
-    'Sort By Most Recent',
-    'Recent Collections',
-    'A-Z',
-    'Z-A',
-    'Price ↓',
-    'Price ↑',
-  ];
-  const AllFilter = [
-    'All Filter',
-    'Price',
-    'Size',
-    'Color',
-    'Reviews',
-    'Brand',
-  ];
+const DropDownViewList = ({
+  selectCategory,
+  selectSortBy,
+  selectAllFilter,
+}) => {
+  const [categories, setCategories] = useState([]);
+  const fetchProducts = () => {
+    fetch(`${apiURL()}/categories/`)
+      .then((res) => res.json())
+      .then((item) => {
+        setCategories(item);
+      });
+  };
+  useEffect(() => {
+    fetchProducts();
+  }, []);
+
+  const Categories = categories.map(
+    (element) =>
+      element.name.charAt(0).toUpperCase() +
+      element.name.slice(1).toLowerCase(),
+  );
+
+  const SortBy = ['Recent Collections', 'Alphabetically', 'Price ↓', 'Price ↑'];
+  const AllFilter = ['Price', 'Size', 'Color', 'Reviews', 'Brand'];
 
   return (
     <div className="list-view">
-      <DropDownView lable="Categories" options={Categories} />
-      <DropDownView lable="Sort By" options={SortBy} />
-      <DropDownView lable="All Filter" options={AllFilter} />
+      <DropDownView
+        lable="Categories"
+        options={Categories}
+        select={selectCategory}
+      />
+      <DropDownView
+        lable="Sort By Most Recent"
+        options={SortBy}
+        select={selectSortBy}
+      />
+      <DropDownView
+        lable="All Filter"
+        options={AllFilter}
+        select={selectAllFilter}
+      />
     </div>
   );
 };
+DropDownViewList.propTypes = {
+  selectCategory: PropTypes.func.isRequired,
+  selectSortBy: PropTypes.func.isRequired,
+  selectAllFilter: PropTypes.func.isRequired,
```

Comment:
These are event handlers, so they are called **when** something happens but they don't decide **what** happens. 

I would advice naming them `onCategorySelected` instead, so that when this component is used it looks like this:
`<DropDownViewList onCategorySelected={setCategory} />`

Reply:
I have change useState name according to feedback.